### PR TITLE
Add unit test case for pipelines/k8s file

### PR DESCRIPTION
**Pull request by ppadti**
Closes: [RHOAIENG-2177](https://issues.redhat.com/browse/RHOAIENG-2177)

## Description
Unit test case for: frontend/src/api/pipelines/k8s.ts

## How Has This Been Tested?
`npm run test`

## Test Impact
Coverage:
Screenshot from 2024-02-06 01-13-58

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by pnaik1 — CHANGES_REQUESTED** [review]

File: frontend/src/api/pipelines/__tests__/k8s.spec.ts [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        name: 'ds-pipeline-pipelines-definition',
+        ns: 'namespace',
+        queryParams: {},
+      },
+    });
+  });
+});
+
+describe('createPipelinesCR', () => {
+  const DSPipelinemock: DSPipelineKind = {
```
[the remark this answers is not in the document]
Comment (ppadti) [reply frontend/src/api/pipelines/__tests__/k8s.spec.ts:100]: yes. In `createPipelinesCR` it is removing some properties from the data that we pass. I don't want to modify the `mockDataSciencePipelineApplicationK8sResource ` while asseting for this test case. So I created one.

[the remark this answers is not in the document]
Comment (christianvogt) [reply frontend/src/api/pipelines/__tests__/k8s.spec.ts:100]: I'm confused. It looks to be exactly the same except for the status. Can you not reuse the existing mock util but delete the status. If it's not about the status, can you elaborate on the need for this mock and not the existing one.

[the remark this answers is not in the document]
Comment (ppadti) [reply frontend/src/api/pipelines/__tests__/k8s.spec.ts:100]: yes, reason to create a new is for status. I thought of not deleting the status from the mock, so created new one. If you want me to reuse that, I will update this test case.

[the remark this answers is not in the document]
Comment (christianvogt) [reply frontend/src/api/pipelines/__tests__/k8s.spec.ts:100]: We have an issue to improve how our mock functions are done: https://issues.redhat.com/browse/RHOAIENG-2412
This is one of those cases where we want to use the mock but configure it to omit the status.

[the remark this answers is not in the document]
Comment (ppadti) [reply frontend/src/api/pipelines/__tests__/k8s.spec.ts]: Updated the test. Ready for review again.

**Pull request merged**